elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,250 @@
+
+
+- The execution of creating an encrypted FileSystem realm would be as follows
+    * Create a CredentialStore (used to store the SecretKey). This would create a credential store and store a secret key inside it with an alias "key"
+    * Then the user would create a FileSystemRealm and pass in the reference of the credential store as well as passing in the alias of the secret key defined in the Elytron subsystem.
+- It will be necessary to modify the `FileSystemRealmDefinition` to incorporate the option to add a path to the credential store as well as the alias for the SecretKey if the user want's to encrypt their FileSystemRealm. This would go under the `RealmAddHandler` class in the `performRuntime()` method. Also will need to add a `SimpleAttributeDefinition` to pass in a SecretKey into the FileSystemSecurityRealm builder. Need to also update the `ElytronDescriptionConstants` with the constants containing encryption information too.
+
+=== Filesystem Integrity
+In the FileSystemSecurityRealm, a method will be added to generate the signature for any given identity file. Another method will be added to verify the signature of a given file.
+
+If a signature already exists for the file before updating the data in the file, it will first be verified to ensure the contents match what is expected, after which the contents will be updated and a new signature will be generated.
+
+Anytime an identity file is read, it will first run the hash checksum verifier, if that results in false, then the verify-signature method is run before proceeding any further. This is to ensure the file has not been tampered with since the last authorized write.
+
+The accepted xml namespaces will also be updated to accept `XMLSignature.XMLNS` that has the value of `http://www.w3.org/2000/09/xmldsig#`.
+
+== Test Plan
+Elytron subsystem filesystem-realm tests will be added. Tests will be added to the Elytron testsuite and the Elytron subsystem tests to ensure that the filesystem always checks the integrity of each identity and passes every time when a keystore reference and an alias to the keypair is provided.
+
+== Community Documentation
+Documentation will be added in the "FileSystem Security Realm" section under elytron/components in the WildFly documentation to indicate that it is possible to encrypt and verify integrity of the filesystem-realm during the creation.
+
+== Release Note Content
+Support for adding filesystem realm encryption support with the use of a SecretKey.
+
+It is now possible to encrypt identities that are stored on the local filesystem when using an Elytron filesystem-realm.
+
+It is also possible to encrypt a filesystem realm that already exists to be compatible with these new changes.
+
+Support for enabling filesystem integrity support using an asymmetric KeyPair.
```

Comment:
This sentence should be removed for now (since this won't be part of 26.1).

Reply:
Removed